src/code/z_player_lib.c

```diff
@@ -1484,8 +1507,46 @@ s32 Player_GetEnvTimerType(PlayState* play) {
     return envTimerType + 1;
 }
 
-void func_80124420(Player* player);
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124420.s")
+void Player_UpdateBunnyEarsKinematics(Player* player) {
+    s32 pad;
+    s16 sp2C;
+    s16 sp28;
+    s16 sp26;
+
+    D_801F59D0.unk_6 -= D_801F59D0.unk_6 >> 3;
```

Comment:
You can document this struct from my OoT PR, don't think that was the controversial part.